modules/web/src/app/cluster/details/cluster/template.html

```diff
@@ -83,37 +83,50 @@
         </button>
         <mat-menu #menu="matMenu"
                   class="km-provider-edit-settings">
-          <button mat-menu-item
-                  (click)="editCluster()"
-                  [disabled]="!isClusterRunning || !isEditEnabled()">
-            <span>Edit Cluster</span>
-          </button>
-          <button mat-menu-item
-                  (click)="editProviderSettings()"
-                  *ngIf="!cluster.spec.cloud.bringyourown"
-                  [disabled]="!isClusterRunning || !isEditEnabled()">
-            <span>Edit Provider</span>
-          </button>
-          <button mat-menu-item
-                  (click)="editSSHKeys()"
-                  *ngIf="cluster.spec.enableUserSSHKeyAgent"
-                  [disabled]="!isSSHKeysEditEnabled()">
-            <span>Manage SSH keys</span>
-          </button>
-          <button mat-menu-item
-                  (click)="revokeToken()"
-                  [disabled]="!isRevokeTokenEnabled()">
-            <span>Revoke Token</span>
-          </button>
+          <span [matTooltip]="!isEditEnabled() ? 'This action is not available for viewer role users.' : ''">
+            <button mat-menu-item
+                    [ngClass]="!isEditEnabled() ? 'remove-hover' : ''"
```

Comment:
```suggestion
                    [ngClass]="{'remove-hover': !isClusterRunning || !isEditEnabled()}"
```
Please also update other changes below to use this `ngClass` syntax and also add `!isClusterRunning` condition where applicable.